activerecord/lib/active_record/suppressor.rb

```diff
@@ -0,0 +1,34 @@
+module ActiveRecord
+  module Suppressor
+    extend ActiveSupport::Concern
+
+    module ClassMethods
+      # Suppresses creation of events for the duration of the block.
+      def suppress(&block)
+        SuppressorRegistry.suppressed = true
+        yield
+      ensure
+        SuppressorRegistry.suppressed = false
+      end
+    end
+
+    # Ignore saving events if we're in suppression mode.
+    def save!(*args)
+      SuppressorRegistry.suppressed ? self : super
+    end
+
+    # This is a thread locals registry for suppressor settings.
+    #
+    # See the documentation of <tt>ActiveSupport::PerThreadRegistry</tt>
+    # for further details.
+    class SuppressorRegistry
```

Comment:
This class isn't for public use, add a `# :nodoc:` to the end of the line. Then remove the documentation about looking the per thread registry.